Enyim.Caching/ZSTDCompression.cs

```diff
@@ -0,0 +1,58 @@
+using ZstdSharp;
+using System;
+using Microsoft.Extensions.Logging;
+using Enyim.Caching;
+
+public static class ZSTDCompression
+{
+    /// <summary>
+    /// Compresses the given data using ZSTD.
+    /// </summary>
+    /// <param name="data">The byte array to compress.</param>
+    /// <returns>Compressed data as an ArraySegment of bytes.</returns>
+    public static ArraySegment<byte> Compress(ArraySegment<byte> data, ILogger<MemcachedClient> _logger)
```

Comment:
It would be better if we create a separate logger for ZSTDCompression, instead of using Memcached Client's logger. Passing the same logger might create unnecessary confusions

Reply:
Discussed